source/slang/slang-ir-layout.h

```diff
@@ -56,14 +56,25 @@ struct IRSizeAndAlignment
 
 struct IRTypeLayoutRules
 {
+private:
+    virtual IRSizeAndAlignment alignCompositeElement(IRSizeAndAlignment elementSize) = 0;
+
 public:
     IRTypeLayoutRuleName ruleName;
-    virtual IRSizeAndAlignment alignCompositeElement(IRSizeAndAlignment elementSize) = 0;
+    virtual IRSizeAndAlignment alignCompositeElementOfNonAggregate(IRSizeAndAlignment elementSize)
```

Comment:
Need comments here explaining when this function is called, and what it attempts to achieve.